### PR TITLE
Add tiptap folder info

### PR DESCRIPTION
**Pull request by BoDonkey**
## Summary

*Summarize the changes briefly, including which issue/ticket this resolves. If it closes an existing Github issue, include "Closes #[issue number]"*
The `asset` module reference page did not include a mention of the `ui/apos/tiptap-extensions` folder that can be included with any module. The PR adds this info. Closes PRO-3074.

## What are the specific steps to test this change?

*For example:*
> 1. Run the website and log in as an admin
> 2. Open a piece manager modal and select several pieces
> 3. Click the "Archive" button on the top left of the manager and confirm that it should proceed
> 4. Check that all pieces have been archived properly

1. Build the documentation
2. Navigate to `/reference/modules/asset.html#build`

## What kind of change does this PR introduce?
*(Check at least one)*

- [ ] Bug fix
- [ ] New feature
- [ ] Refactor
- [X] Documentation
- [ ] Build-related changes
- [ ] Other

## Make sure the PR fulfills these requirements:

- [X] It includes a) the existing issue ID being resolved, b) a convincing reason for adding this feature, or c) a clear description of the bug it resolves
- [ ] The changelog is updated
- [ ] Related documentation has been updated
- [ ] Related tests have been updated

If adding a new feature without an already open issue, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

### CODE REVIEW

**Review by boutell — CHANGES_REQUESTED** [review]

File: docs/reference/modules/asset.md [hunk 1]
```diff
@@ -36,7 +36,7 @@ When the `APOS_UPLOADFS_ASSETS` environment variable is present, this optional p
...
...any new Vue components, an `apps` folder that takes any additional admin-facing JavaScript, and a `tiptap-extensions` folder that contains any tiptap-extensions used within the admin UI. During the build process, code located in the `ui/apos` subdirectory of any module is automatically detected and incorporated. Assets in the `components` folder are registered automatically by name and do not need to be imported. Unlike assets in the `ui/src` folder where only the `index.js` file is an entry point, all files in the `ui/apos/apps` folder are entry points. If this behavior is undesirable any files that should not be entry points can be placed into a sub-directory and imported into the main entry point file. See the [`@apostrophecms/login` module](https://github.com/apostrophecms/apostrophe/tree/main/modules/%40apostrophecms/login) for an example, including import of the Vue library within the [`AposLogin.js` file](https://github.com/apostrophecms/apostrophe/blob/main/modules/%40apostrophecms/login/ui/apos/apps/AposLogin.js). Customizing the admin UI is covered in-depth in the [guide documentation](https://v3.docs.apostrophecms.org/guide/custom-ui.html#customizing-the-user-interface).
```
[the remark this answers is not in the document]
Comment (BoDonkey) [reply docs/reference/modules/asset.md]: Added and restructured.

File: docs/reference/modules/asset.md [hunk 2]
```diff
@@ -36,7 +36,11 @@ When the `APOS_UPLOADFS_ASSETS` environment variable is present, this optional p
...
+
+During the build process, code located in the `ui/apos` subdirectory of any module is automatically detected and incorporated. Assets in the `components` folder are registered automatically by name and do not need to be imported. Unlike assets in the `ui/src` folder where only the `index.js` file is an entry point, all files in the `ui/apos/apps` folder are entry points. If this behavior is undesirable any files that should not be entry points can be placed into a sub-directory and imported into the main entry point file. See the [`@apostrophecms/login` module](https://github.com/apostrophecms/apostrophe/tree/main/modules/%40apostrophecms/login) for an example, including import of the Vue library within the [`AposLogin.js` file](https://github.com/apostrophecms/apostrophe/blob/main/modules/%40apostrophecms/login/ui/apos/apps/AposLogin.js). Customizing the admin UI is covered in-depth in the [guide documentation](https://v3.docs.apostrophecms.org/guide/custom-ui.html#customizing-the-user-interface).
```
Comment (boutell) [inline docs/reference/modules/asset.md:41]: Assets in the components folder are automatically registered as Vue components

File: docs/reference/modules/asset.md [hunk 3]
```diff
@@ -36,7 +36,11 @@ When the `APOS_UPLOADFS_ASSETS` environment variable is present, this optional p
...
+
+The code in the `tiptap-extensions` folder is typically used to extend the `rich-text-widget`, which uses the [tiptap WYSIWYG editor](https://tiptap.dev/) at its core. This editor has a wealth of pre-made [extensions](https://tiptap.dev/extensions) that could be incorporated.
```
Comment (boutell) [inline docs/reference/modules/asset.md:43]: Current language suggests it's loosey-goosey but it's more specific and only the right stuff will work, so:

"Every file in the `tiptap-extensions` folder must be a tiptap WYSIWYG rich text editor extension, written in that format. Every such extension will be loaded and made available for potential activation by the developer in the rich text widget's editing toolbar."

**Review by boutell — CHANGES_REQUESTED** [review]

**Comment by boutell** [comment]
So it's rather like the components folder, only a Vue component makes sense there.

**Pull request merged**